src/pudl/metadata/resources/__init__.py

```diff
@@ -288,11 +288,30 @@
             "phone_number_2",
             "data_maturity",
         ],
+        "map_cols_dict": {
+            "owner_utility_id_eia": "utility_id_eia",
+            "owner_zip_code": "zip_code",
+            "owner_city": "city",
+            "owner_name": "utility_name_eia",
+            "owner_state": "state",
+            "owner_street_address": "street_address",
+        },
     },
 }
 """Columns kept for either entity or annual EIA tables in the harvesting process.
 
-For each entity type (key), the ID columns, static columns, and annual columns,
+For each entity type (key), the ID columns, static columns, annual columns, and mapped
+columns.
+
+Mapped columns allow for harvesting the same entity ID / value relationship
+from multiple columns in the same input dataframe. This is useful if an entity has a
+relationship with another entity of the same kind, for example owner and operator utilities
+showing up in the same ownerhip table records. The mapped column dictionary maps from
+column names of the second group of entity ID / value columns to the standard names
+for the entity ID / values columns.
+
+Mapped column dictionaries must include all columns that are to be harvested
+for that relationship, even if a column name maps to itself.
```

Comment:
I tried to explain the change in this docstring, but maybe this doesn't make sense. Not sure what the best explanation of this behavior is that would generalize to potential future scenarios.

Reply:
I'm not understanding the last bit:
> Mapped column dictionaries must include all columns that are to be harvested
for that relationship, even if a column name maps to itself.

Reply:
I think the example I was thinking of wouldn't actually come up, because it would require a table input that's not so well normalized/clean. I think I can take this part out, but this is what I was originally thinking:

- There are the columns `owner_utility_id_eia`, `utility_id_eia`, but the owner and operator utility share a column, maybe `state` or something. Maybe this example won't ever come up, but perhaps we're only looking at CA utilities so there's one state column shared between owner and operator
- In this case, you'd want to harvest `state` for both the owner and operator utility, but with the owner utility you don't need to rename the `state` column, so you map it to itself `state:state`.